sale.py

```diff
@@ -335,3 +278,30 @@ def _get_amount_to_checkout(self):
         per their requirement
         """
         return self.amount_to_receive
+
+    def _get_email_template_context(self):
+        """
+        Update context
+        """
+        context = super(Sale, self)._get_email_template_context()
+
+        context.update({
+            'url_for': lambda *args, **kargs: url_for(*args, **kargs),
```

Comment:
I think you can pass customer_name from here only. according to request context, as this is required in all cases.